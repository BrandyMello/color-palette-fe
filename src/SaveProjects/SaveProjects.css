.save_project_form {
  margin-left: 30%;
  margin-top: 30px;
}

.project_input {
  display: inline-block;
  font-size: 18px;
  width: 450px;
  height: 48px;
}

.add_project_button {
  background-color: #808080;
  color: white;
  font-size: 16px;
  font-weight: bold;
  display: inline-block;
  width: 174px;
  height: 54px;
}