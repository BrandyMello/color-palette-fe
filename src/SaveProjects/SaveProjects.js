import React, { Component } from "react";
import "./SaveProjects.css";

class SaveProjects extends Component {
  constructor() {
    super();
    this.state = {
      projectName: ""
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  clearInput = () => {
    this.setState({ projectName: "" });
  };

  render() {
    return (
      <form className="save_project_form">
        <input
          className="project_input"
          type="text"
          id="name"
          name="projectName"
          placeholder="New Project Name"
          value={this.state.projectName}
          onChange={this.handleChange}
          onClick={() => this.clearInput()}
        />
        <button
          className="add_project_button"
          name="newProject"
          type="submit"
          onClick={e =>
            this.props.saveNewProject({ name: this.state.projectName }, e)
          }
        >
          Create New Project
        </button>
      </form>
    );
  }
}

export default SaveProjects;
