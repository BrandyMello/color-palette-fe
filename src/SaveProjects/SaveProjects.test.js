import React from "react";
import { shallow } from "enzyme";
import SaveProjects from "./SaveProjects";

describe("SaveProjects", () => {
  let wrapper;
  let mockProjects;

  beforeEach(() => {
    mockProjects = [{ name: "Autumn", id: 1 }, { name: "Seabreeze", id: 2 }];

    wrapper = shallow(
      <SaveProjects projects={mockProjects} saveNewProject={jest.fn()} />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should update state when handleChange is called", () => {
    const mockEvent = {
      target: { name: "projectName", value: { name: "Fall", id: 3 } }
    };
    const expected = { name: "Fall", id: 3 };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state("projectName")).toEqual(expected);
  });

  it("should reset state when clearInput is called", () => {
    const expected = { projectName: "" };

    wrapper.instance().setState({ projectName: "Fall" });

    wrapper.instance().clearInput();

    expect(wrapper.state()).toEqual(expected);
  });

  it("should run handleChange when the inputs detect a change", () => {
    const mockEvent = {
      target: { name: "projectName", value: { name: "Fall", id: 3 } }
    };

    wrapper.instance().handleChange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.update();
    wrapper.find('[name="projectName"]').simulate("change", mockEvent);

    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent);
  });
});
