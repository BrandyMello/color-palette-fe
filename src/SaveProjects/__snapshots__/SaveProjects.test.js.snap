// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveProjects should match snapshot 1`] = `
<form
  className="save_project_form"
>
  <input
    className="project_input"
    id="name"
    name="projectName"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="New Project Name"
    type="text"
    value=""
  />
  <button
    className="add_project_button"
    name="newProject"
    onClick={[Function]}
    type="submit"
  >
    Create New Project
  </button>
</form>
`;
